extensions/git-base/src/remoteSource.ts

```diff
@@ -70,12 +89,17 @@ class RemoteSourceProviderQuickPick {
 			this.quickpick!.items = [{ label: l10n.t('{0} Error: {1}', '$(error)', err.message), alwaysShow: true }];
 			console.error(err);
 		} finally {
-			this.quickpick!.busy = false;
+			if (!this.isDisposed) {
+				this.quickpick!.busy = false;
+			}
 		}
 	}
 
 	async pick(): Promise<RemoteSource | undefined> {
 		await this.query();
+		if (this.isDisposed) {
+			return;
+		}
 		const result = await getQuickPickResult(this.quickpick!);
```

Comment:
this function has a few events that aren't getting disposed.

Reply:
Good find, thanks!